tests/binary_agreement.rs

```diff
@@ -1,108 +1,137 @@
 #![deny(unused_must_use)]
-//! Tests of the Binary Agreement protocol. Only one proposer instance
-//! is tested. Each of the nodes in the simulated network run only one instance
-//! of Binary Agreement. This way we only test correctness of the protocol and not
-//! message dispatch between multiple proposers.
+//! Tests of the Binary Agreement protocol
+//!
+//! Each of the nodes in the simulated network runs only one instance of Binary Agreement for node
+//! `0`, the proposer. In real applications there would be multiple proposers in the
+//! network. However, having only one proposer allows to test correctness of the protocol rather
+//! than message dispatch between multiple proposers.
 //!
 //! There are three properties that are tested:
 //!
-//! - Agreement: If any correct node outputs the bit b, then every correct node outputs b.
+//! - Agreement: If any correct node outputs the bit `b`, then every correct node outputs `b`.
 //!
 //! - Termination: If all correct nodes receive input, then every correct node outputs a bit.
 //!
-//! - Validity: If any correct node outputs b, then at least one correct node received b as input.
-//!
-//! TODO: Implement adversaries and send BVAL messages at different times.
+//! - Validity: If any correct node outputs `b`, then at least one correct node received `b` as
+//! input.
 
-extern crate env_logger;
+extern crate failure;
 extern crate hbbft;
-extern crate log;
+extern crate integer_sqrt;
+extern crate proptest;
 extern crate rand;
-extern crate rand_derive;
-extern crate serde_derive;
-extern crate threshold_crypto as crypto;
+extern crate threshold_crypto;
 
-mod network;
+pub mod net;
```

Comment:
That shouldn't be `pub`.

Reply:
There are lots of "method is never used" lints otherwise in the `net` module.

Reply:
Right; some of them are already marked as `#[allow(unused)]`—maybe we should apply that attribute to the whole `net` module instead (either inside it, or at the `use` statements)? That would make the reason clearer than the `pub`.

Reply:
One option would also be to make the testing framework a proper crate and just import that; I've done this on another project.

It's a bit more work, but it also does not require importing a million different crates on each test cases - right now, whenever I add a dependency to the `net` testing framework, I also need to add one in every test case.